boot-speed/clouds/measure-cloud.py

```diff
@@ -200,43 +215,65 @@ def gen_datadirname(metadata):
     """ Generate a standardized measurement directory (and tarball) name """
     yyyymmdd = metadata['date']
     cloud = metadata['instance']['cloud']
+    inst_type = metadata['instance']['instance_type']
     release = metadata['instance']['release']
 
-    datadir = Path(cloud + "-" + release + "_" + yyyymmdd)
+    datadir = cloud + "-" + inst_type + "-" + release + "_" + yyyymmdd
     return datadir
 
 
-def main(cloud, release, instances, reboots):
-    if cloud not in known_clouds:
+def main():
+    args = parse_args()
+
+    if args.cloud not in known_clouds:
         print('Unknown cloud provider:', cloud)
         sys.exit(1)
 
+    if args.cloud == 'ec2':
+        instspec = EC2Instspec(args.release)
+        if args.inst_type:
+            instspec.inst_type = args.inst_type
```

Comment:
Should these be parameters on `__init__` instead? I can't see anywhere that we use them as class attributes.